src/effects/VST/VSTEffect.cpp

```diff
@@ -911,6 +911,139 @@ bool VSTEffect::InitializePlugin()
    return true;
 }
 
+
+bool VSTEffectWrapper::TransferSettingsContents
+(
+   VSTEffectSettings& src,
+   VSTEffectSettings& dst,
+   bool doMove,
+   bool doMerge
+)
+{
+   dst.mNumParams = src.mNumParams;
+   dst.mUniqueID  = src.mUniqueID;
+   dst.mVersion   = src.mVersion;
+
+   assert(dst.mParamsMap.size() == src.mParamsMap.size());
+
+   // Do the chunk first
+   if (src.mChunk)
+   {
+      dst.mChunk = src.mChunk;
+      if (doMove)
+      {
+         src.mChunk = std::nullopt;
+      }
+   }
+   else if(!doMerge)
+   {
+      dst.mChunk = std::nullopt;
+   }
+   
+
```

Comment:
All the stuff above with chunks and the rest does not belong in the Message object.

VST3EffectSettings needs a base class, or a member, which contains only the map.

TransferSettingsContents should only be concerned with two maps.

Reply:
Well, VST3EffectSettings already has a member that (in a way) contains only the map - it's the map itself. I could just remove from this method all lines that reference what is not the map. Would that be fine?

Reply:
ok, the map itself can be the argument type passed to these functions.